src/test/java/net/masterthought/cucumber/ReportParserTest.java

```diff
@@ -88,4 +90,145 @@ public void parseJsonResults_OnNoExistingFile_ThrowsException() {
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Contents_Order() {
+
+        // given
+        initWithProperties(SAMPLE_TWO_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
+
+        // then
+        assertThat(classifications).hasSize(3);
+        assertThat(classifications.get(0).getKey()).isEqualTo("NodeJsVersion");
+        assertThat(classifications.get(1).getKey()).isEqualTo("Proxy");
+        assertThat(classifications.get(2).getKey()).isEqualTo("NpmVersion");
+
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Duplicates() {
+
+        // given
+        initWithProperties(DUPLICATE_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
+
+        // then
+        assertThat(classifications).hasSize(1);
+        assertThat(classifications.get(0).getKey()).isEqualTo("BaseUrl_QA");
+        assertThat(classifications.get(0).getValue()).isEqualTo("[Internal=https://internal.test.com, External=https://external.test.com]");
+
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Special_Characters() {
+
+        // given
+        initWithProperties(SPECIAL_CHARACTERS_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
+
+        // then
+        assertThat(classifications).hasSize(6);
+        assertThat(classifications).containsExactly(
+                entry("website","https://en.wikipedia.org/"),
+                entry("language", "English"),
+                entry("message", "Welcome to Wikipedia!"),
+                entry("key with spaces", "This is the value that could be looked up with the key \"key with spaces\"."),
+                entry("tab", "\t"),
+                entry("path", "c:\\wiki\\templates")
+        );
+
+    }
+
+    @Test
+    public void parsePropertyFiles_OnInvalidFilePath_ThrowsException() {
+
+        // given
+        final String invalidFile = "?on-invalid-file-path.properties";
+        initWithProperties(EMPTY_PROPERTIES);
+        propertyReports.add(invalidFile);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // then
+        thrown.expect(ValidationException.class);
+        thrown.expectMessage(containsString(invalidFile));
+        thrown.expectMessage(endsWith("doesn't exist or the properties file is invalid!"));
```

Comment:
error message could point out which properties files is invalid

Reply:
I am doing this already tho?
```
throw new ValidationException(String.format("File '%s' doesn't exist or the properties file is invalid!", file), e);
```
is been thrown by the code and we are checking for presence of the filename

```
thrown.expectMessage(containsString(invalidFile));
thrown.expectMessage(endsWith("doesn't exist or the properties file is invalid!"));
```

Reply:
Makes sense